File: 169.majority-element.py
#
# @lc app=leetcode id=169 lang=python3
#
# [169] Majority Element
#

from collections import defaultdict
from typing import List
import statistics

# @lc code=start


class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        return int(statistics.median(nums))

# @lc code=end

File: 169.majority-element.cpp
/*
 * @lc app=leetcode id=169 lang=cpp
 *
 * [169] Majority Element
 */

#include <vector>

using namespace std;

// @lc code=start
class Solution {
public:
    int majorityElement(const vector<int> &nums) {
        int candidate = 0, count = 0;

        for (int num : nums) {
            if (count == 0) {
                candidate = num;
            }
            count += (num == candidate) ? 1 : -1;
        }

        return candidate;
    }
};
// @lc code=end

